src/change_stream/mod.rs

```diff
@@ -227,11 +220,47 @@ where
     T: DeserializeOwned + Unpin + Send + Sync,
 {
     fn poll_next_in_batch(&mut self, cx: &mut Context<'_>) -> Poll<Result<BatchValue>> {
```

Comment:
This is essentially the manual state machine version of `SessionChangeStream::next_if_any`.  Hopefully at some point Rust gets reasonable first-class syntax for streams.

I considered trying to write this by wrapping an async function in a stream adaptor for clarity, but that ended up needing to be self-referential.